engine/src/main/java/org/destinationsol/game/console/commands/InvincibleCommandHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game.console.commands;
+
+import org.destinationsol.game.Console;
+import org.destinationsol.game.Hero;
+import org.destinationsol.game.SolGame;
+import org.destinationsol.game.console.ConsoleInputHandler;
+import org.destinationsol.game.ship.SolShip;
+
+/**
+ * A command used to make player invincible
+ */
+public class InvincibleCommandHandler implements ConsoleInputHandler {
+
+    private Hero hero;
+    private SolGame game;
+    private SolShip solShip;
+
+    public InvincibleCommandHandler(Hero hero, SolGame game, SolShip solShip) {
+        this.hero = hero;
+        this.game = game;
+        this.solShip = solShip;
```

Comment:
Herein lies a problem, that is, solShip is recreated every time it exits from a starport, and thus, the moment a hero enters the starport, this solShip object loses its purpose